modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,49 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
 
-				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
+				var images = document.querySelectorAll( 'img.wp-image-' + media[ i ].id ),
+					media_sources = media[ i ].media_details.sources;
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
-					var srcset = images[j].getAttribute( 'srcset' );
+
+					if ( ! media_sources || ! media_sources['image/jpeg'] ) {
+						continue;
+					}
+
+					var srcset = images[ j ].getAttribute( 'srcset' ),
+						sizes = media[ i ].media_details.sizes,
```

Comment:
Let's assign this variable outside of this `for` loop for a bit more efficiency. It is the same for every value in `images`, so we can set this variable once, e.g. in the same place where you're setting `media_sources`.